packages/Modal/src/Modal.js

```diff
@@ -0,0 +1,149 @@
+/* eslint-disable jsx-a11y/no-static-element-interactions */
+
+import React from "react";
+import ReactDOM from "react-dom";
+import PropTypes from "prop-types";
+import classNames from "classnames";
+import uuid from "uuid/v4";
+import stylers from "@paprika/stylers";
+import { ShirtSizes } from "@paprika/helpers/lib/customPropTypes";
+import { contentStyles, modalStyles, frameStyles, overlayStyles } from "./Modal.styles";
+
+const propTypes = {
+  /** Aria title of dialog for accessibility. */
+  ariaLabel: PropTypes.string.isRequired,
+
+  /** Content for the Modal. */
+  children: PropTypes.node.isRequired,
+
+  /** Custom classes for the root element of the Modal. */
+  className: PropTypes.string,
+
+  /** If the Modal includes an overlay for content below it. */
+  hasOverlay: PropTypes.bool,
+
+  /** Controls the visibility of the Modal. */
+  isOpen: PropTypes.bool,
+
+  /** Function to call when "esc" key is pressed. */
+  onEscKey: PropTypes.func,
+
+  /** Function to call when overlay is clicked. */
+  onOverlayClick: PropTypes.func,
+
+  /** Exposes a reference to the Modal's portal element. */
+  portalRef: PropTypes.func,
+
+  /** Makes the content of the modal scrollable, but as a side effect prevents any content (e.g. MultiSelects) from
+  overflowing. If you need this on, you may want to consider using a different component instead of this one. */
+  isScrollable: PropTypes.bool,
+
+  /** Width of the Modal. */
+  width: PropTypes.oneOf(ShirtSizes.DEFAULT),
+
+  /** Optional override of z-index. */
+  zIndex: PropTypes.number,
+};
+
+const defaultProps = {
+  className: "",
+  hasOverlay: true,
+  isOpen: false,
+  isScrollable: false,
+  onEscKey: () => {},
+  onOverlayClick: () => {},
+  portalRef: () => {},
+  width: "medium",
+  zIndex: null,
+};
+
+let modalRoot;
+
+class Modal extends React.Component {
+  constructor(props) {
+    super(props);
+    this.modalContainer = null;
+    this.ariaId = uuid();
+
+    // TODO: make this optional
+    if (!modalRoot) {
+      modalRoot = document.createElement("div");
+      document.body.appendChild(modalRoot);
+    }
+
+    this.el = document.createElement("div");
+  }
+
+  componentDidMount() {
+    document.addEventListener("keydown", this.handleKeyDown, false);
+    modalRoot.appendChild(this.el);
+  }
+
+  componentDidUpdate() {
+    this.renderModal();
```

Comment:
How does it even work? 🤔

Reply:
<magic.gif>

Reply:
![Screen Shot 2019-07-19 at 10 59 24 AM](https://user-images.githubusercontent.com/3008260/61555499-6c075b00-aa14-11e9-8a73-f6750175eec3.png)
@mikrotron may have the answer :P

Reply:
Naîveté? It doesn't actually depend on this to work, does it? Can you just remove it?

Reply:
This will get refactored though anyway?

Reply:
Right it will end up being removed